src/libraries/System.Private.CoreLib/src/System/Reflection/SignatureConstructedGenericType.cs

```diff
@@ -30,6 +30,7 @@ internal SignatureConstructedGenericType(Type genericTypeDefinition, Type[] type
         protected sealed override bool IsArrayImpl() => false;
         protected sealed override bool IsByRefImpl() => false;
         public sealed override bool IsByRefLike => _genericTypeDefinition.IsByRefLike;
+        public sealed override bool IsEnum => false;
```

Comment:
This is not correct. Generic enums are a thing. This should return `_genericTypeDefinition.IsEnum` like before the last commit.

For example, the following should print `true`, but it incorrectly prints `false` with this change.

```
Console.WriteLine(Type.MakeGenericSignatureType(typeof(GenericType<>.GenericEnum), typeof(object)).IsEnum);

class GenericType<T> {
   public enum GenericEnum {
   }
}
```

Reply:
Sure that makes sense since the nested type could use the parent's generic parameter although for an enum that isn't possible. I'll update this and add a test.